signature/src/image/mod.rs

```diff
@@ -75,53 +77,25 @@ impl Image {
         }
     }
 
-    pub fn signatures(&mut self) -> Result<Vec<Vec<u8>>> {
+    pub fn signatures(&mut self, scheme: &str) -> Result<Vec<Vec<u8>>> {
         if self.signatures.is_empty() {
-            return self.internal_signatures();
+            return self.internal_signatures(scheme);
         }
 
         Ok(self.signatures.clone())
     }
 
-    fn internal_signatures(&mut self) -> Result<Vec<Vec<u8>>> {
-        // Get image digest (manifest digest)
-        let image_digest = if !self.manifest_digest.is_empty() {
-            self.manifest_digest.clone()
-        } else if let Some(d) = self.reference.digest() {
-            Digest::try_from(d)?
-        } else {
-            return Err(anyhow!("Missing image digest"));
-        };
-
-        // Format the sigstore name: `image-repository@digest-algorithm=digest-value`.
-        let sigstore_name = signatures::format_sigstore_name(&self.reference, image_digest);
-
-        // If the registry support `X-Registry-Supports-Signatures` API extension,
-        // try to get signatures from the registry first.
-        // Else, get signatures from "sigstore" according to the sigstore config file.
-        if self.registry_xrss_api_support {
-            // TODO: Add get signatures from registry X-R-S-S API extension.
-            return Err(anyhow!(
-                "Now not support get signatures from registry X-R-S-S API extension."
-            ));
-        } else if self.sigstore_base_url.is_none() {
-            let sigstore_config =
-                signatures::SigstoreConfig::new_from_configs(signatures::SIGSTORE_CONFIG_DIR)?;
-            if let Some(base_url) = sigstore_config.base_url(&self.reference)? {
-                self.set_sigstore_base_url(base_url)?;
+    fn internal_signatures(&mut self, scheme: &str) -> Result<Vec<Vec<u8>>> {
+        match crate::SignatureScheme::from_str(scheme) {
+            Ok(SignatureScheme::SimpleSigning) => {
+                self.signatures = mechanism::simple::get_signatures(self)?;
+            }
+            // TODO: Add more signature mechanism.
```

Comment:
- Is there a known list that should be added?
- A GitHub issue URL would be good here.